tools/cli/factory.go

```diff
@@ -156,9 +164,10 @@ func (b *clientFactory) ensureDispatcher(c *cli.Context) {
 	if shouldUseGrpc {
 		b.hostPort = grpcPort
 	}
-	if addr := c.GlobalString(FlagAddress); addr != "" {
```

Comment:
what is this change for?

Reply:
i think it's because we created addressFlagFunc func(c *cli.Context) string in the clientFactory struct and we used that instead of c.GlobalString